sdk/serialization/azure-json/src/main/java/com/azure/json/implementation/DefaultJsonReader.java

```diff
@@ -87,6 +87,7 @@ private DefaultJsonReader(JsonParser parser, boolean resetSupported, byte[] json
         this.parser = parser;
         this.parser.configure(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS.mappedFeature(), nonNumericNumbersSupported);
         this.resetSupported = resetSupported;
+        this.parser.enable(JsonParser.Feature.ALLOW_COMMENTS);
```

Comment:
Let's enable this more richly by adding an option to configure this in JsonOptions as we'll need this to be supported not only in the `DefaultJsonReader` but implementors of `JsonReader` like our `azure-core-serializer-json-gson` and `azure-core-serializer-json-jackson`.